### PR TITLE
Save itenerary

### PR DESCRIPTION
**Pull request by mgjeffries**
Checkout and serve Locally

1. Verify that the save button is disabled until all attractions are selected
1. Verify that the save button is enabled once all attraction are selected
1. Verify that itineraries are saved to the database.

### CODE REVIEW

File: scripts/itineraries/ItinerarySaveButton.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import { saveItinerary } from "./ItineraryProvider.js";
+
+const contentTarget = document.querySelector(".saveItineraryButtonContainer")
+const eventHub = document.querySelector(".container")
+
+let isParkSelected = false
+let isEaterySelected = false
+let isOdditySelected = false
+let iteneraryData = {
```
Comment (esthermcmahon) [inline scripts/itineraries/ItinerarySaveButton.js]: change to itineraryData

File: scripts/itineraries/ItinerarySaveButton.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  oddityName: "",
+  eateryId: "",
+  eateryName: "",
+}
+
+eventHub.addEventListener("parkSelected", customEvent => {
+  
+  if ( customEvent.detail.parkId !== "" ) {
+    isParkSelected = true
+    iteneraryData.parkId = customEvent.detail.parkId
```
Comment (esthermcmahon) [inline scripts/itineraries/ItinerarySaveButton.js]: itineraryData

File: scripts/itineraries/ItinerarySaveButton.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    iteneraryData.parkName = customEvent.detail.parkName
```
Comment (esthermcmahon) [inline scripts/itineraries/ItinerarySaveButton.js]: itineraryData

File: scripts/itineraries/ItinerarySaveButton.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    render()
+  }
+
+})
+eventHub.addEventListener("odditySelected", customEvent => {
+  
+  if ( customEvent.detail.oddityId !== "" ) {
+    isOdditySelected = true
+    iteneraryData.oddityId = customEvent.detail.oddityId
```
Comment (esthermcmahon) [inline scripts/itineraries/ItinerarySaveButton.js]: itinerary

File: scripts/itineraries/ItinerarySaveButton.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    iteneraryData.oddityName = customEvent.detail.oddityName
```
Comment (esthermcmahon) [inline scripts/itineraries/ItinerarySaveButton.js]: itinerary

File: scripts/itineraries/ItinerarySaveButton.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    render()
+  }
+
+})
+eventHub.addEventListener("eaterySelected", customEvent => {
+  
+  if ( customEvent.detail.eateryId !== "" ) {
+    isEaterySelected = true
+    iteneraryData.eateryId = customEvent.detail.eateryId
```
Comment (esthermcmahon) [inline scripts/itineraries/ItinerarySaveButton.js]: itinerary

**Review by skratz17 — APPROVED** [review]
I am down with this sickness

**Review by Raparker88 — APPROVED** [review]

File: scripts/itineraries/ItinerarySaveButton.js (hunk 1 above)
Comment (mgjeffries) [reply scripts/itineraries/ItinerarySaveButton.js]: Oops. Lots of mispellings in this file. Tried to catch them in an earlier commit, but I guess I fumbled the save and commit process.

**Review by esthermcmahon — APPROVED** [review]

**Pull request merged**